src/hotspot/share/logging/logFileOutput.cpp

```diff
@@ -304,12 +300,33 @@ int LogFileOutput::write(const LogDecorations& decorations, const char* msg) {
   return written;
 }
 
+int LogFileOutput::write(const LogDecorations& decorations, const char* msg) {
+  if (_stream == NULL) {
+    // An error has occurred with this output, avoid writing to it.
+    return 0;
+  }
+
+  LogAsyncFlusher* flusher = LogAsyncFlusher::instance();
+  if (flusher != NULL) {
+    flusher->enqueue(*this, decorations, msg);
+    return 0;
```

Comment:
should return -1 here. be consistent with `write(LogMessageBuffer::Iterator msg_iterator)` below.